src/listensocket.cpp

```diff
@@ -92,7 +92,7 @@ ListenSocket::ListenSocket(std::shared_ptr<ConfigTag> tag, const irc::sockets::s
 	{
 		const std::string permissionstr = tag->getString("permissions");
 		unsigned long permissions = strtoul(permissionstr.c_str(), NULL, 8);
-		if (permissions && permissions <= 07777)
+		if (permissions && permissions <= 7777)
```

Comment:
This is specifically intended to be octal. Octal 7777 is the maximum value for permissions for a file.